src/robot_upstart/providers.py

```diff
@@ -34,6 +34,13 @@
 
 from catkin.find_in_workspaces import find_in_workspaces
 
+def detect_provider():
+    cmd=open('/proc/1/cmdline', 'rb').read().split('\x00')[0]
```

Comment:
Oh, there are a bunch of linter problems right now.

Nevermind. I will merge this and fix them in a batch. Thanks!